src/task/task_id.rs

```diff
@@ -13,15 +13,15 @@ use std::sync::atomic::{AtomicU64, Ordering};
 /// })
 /// ```
 #[derive(Eq, PartialEq, Clone, Copy, Hash, Debug)]
-pub struct TaskId(pub(crate) u64);
+pub struct TaskId(pub(crate) usize);
 
 impl TaskId {
     /// Generates a new `TaskId`.
     pub(crate) fn generate() -> TaskId {
-        static COUNTER: AtomicU64 = AtomicU64::new(1);
+        static COUNTER: AtomicUsize = AtomicUsize::new(1);
```

Comment:
This should still be an `AtomicU64`, I'm afraid. What we can do is use `AtomicCell<u64>` from `crossbeam-utils`.

Reply:
I have changed it as you suggest